front/app/api/permissions_custom_fields/useAddPermissionsCustomField.ts

```diff
@@ -0,0 +1,53 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { CLErrors } from 'typings';
+import fetcher from 'utils/cl-react-query/fetcher';
+import permissionsCustomFieldsKeys from './keys';
+import {
+  IListParameters,
+  IPermissionsCustomField,
+  IPermissionsCustomFieldAdd,
+} from './types';
+
+const addPermissionsCustomField = async (
+  requestBody: IPermissionsCustomFieldAdd
```

Comment:
Minor naming thing, but `action` and `phaseId` etc are not in the `body` so maybe we could just call this props or parameters or something